### PR TITLE
Add RulesMD project for generating markdown from rules info

### PR DESCRIPTION
**Pull request by RobGallo**
#### Describe the change

This utility will be run during the build pipeline to generate documentation about Axe.Windows' rules.

- Add target to run RulesMD after build

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

File: src/RulesMD/MarkdownCreator.cs [hunk 1]
```diff
@@ -0,0 +1,354 @@
+﻿// ------------------------------------------------------------------------------
```
Comment (DaveTryon) [inline src/RulesMD/MarkdownCreator.cs]: Strangely enough, this is passing the license header check in the debug build, but not in the release build. Is it possible to generate this file at build time instead of having it in the repo?

File: src/AxeWindows.sln [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.28307.757
+# Visual Studio Version 16
```
Comment (DaveTryon) [inline src/AxeWindows.sln]: This is stamped for VS2019. Should we make 2019 our official build environment? I like the plan, but it feels like it should be a separate PR (and update the docs that reference VS2017)

File: src/RulesMD/MarkdownCreator.cs (hunk 1 above)
Comment (RobGallo) [reply src/RulesMD/MarkdownCreator.cs]: Yes, we are now preprocessing the templates before the build and writing the output to the obj folder.

File: src/AxeWindows.sln [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
+# Visual Studio Version 15
```
Comment (DaveTryon) [inline src/AxeWindows.sln]: "Version" should not appear here
```suggestion
# Visual Studio 15
```

File: src/RulesMD/Helpers.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+namespace RulesMD
+{
+    static class Helpers
+    {
+        public static string GetStandardName(A11yCriteriaId criteriaId)
+        {
+            switch (criteriaId)
+            {
+                case A11yCriteriaId.AvailableActions:
+                    return "Section 508 502.3.10";
```
Comment (DaveTryon) [inline src/RulesMD/Helpers.cs]: >Section [](start = 28, length = 7)

Can we possibly pull these strings from the RuleSelection assembly? That would give us one less thing to maintain. If the string resources aren't available directly, we could consider adding a public static method to the RuleSelection assembly--pass in an A11yCriteriaId, get back the string description.

File: src/RulesMD/Program.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        private static void Run(CLIOptions options)
+        {
+            var markdownCreator = new MarkdownCreator();
+            var markdown = markdownCreator.TransformText();
+
+            var f = File.CreateText(options.OutputPath);
+            f.Write(markdown);
+            f.Close();
+        }
+} // class
```
Comment (DaveTryon) [inline src/RulesMD/Program.cs]: Please indent this line
```suggestion
    } // class
```

File: src/RulesMD/MarkdownCreator.cs (hunk 1 above)
Comment (DaveTryon) [reply src/RulesMD/MarkdownCreator.cs]: Thanks! That's much nicer

File: src/RulesMD/RulesMD.csproj [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      <Project>{7fe58829-560e-4142-bf81-291a66d3844b}</Project>
+      <Name>Rules</Name>
+    </ProjectReference>
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <Import Project="..\..\build\delaysign.targets" />
+    <Import Project="$(VSToolsPath)\TextTemplating\Microsoft.TextTemplating.targets" />
+<Target Name="GenerateMarkdown" AfterTargets="Build" Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU'">
+  <Exec Command="echo Generate markdown of rules info" />
+  <Exec Command="$(TargetPath) --output $(TargetDir)\axe-windows-rules.md"  />
```
Comment (DaveTryon) [inline src/RulesMD/RulesMD.csproj]: Indents are off
```suggestion
  <Import Project="$(VSToolsPath)\TextTemplating\Microsoft.TextTemplating.targets" />
  <Target Name="GenerateMarkdown" AfterTargets="Build" Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU'">
    <Exec Command="echo Generate markdown of rules info" />
    <Exec Command="$(TargetPath) --output $(TargetDir)\axe-windows-rules.md"  />
```

File: src/RulesMD/RulesMD.csproj [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+      <Name>Core</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Rules\Rules.csproj">
+      <Project>{7fe58829-560e-4142-bf81-291a66d3844b}</Project>
+      <Name>Rules</Name>
+    </ProjectReference>
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <Import Project="..\..\build\delaysign.targets" />
+    <Import Project="$(VSToolsPath)\TextTemplating\Microsoft.TextTemplating.targets" />
```
Comment (DaveTryon) [inline src/RulesMD/RulesMD.csproj]: Indent is off
```suggestion
  <Import Project="$(VSToolsPath)\TextTemplating\Microsoft.TextTemplating.targets" />
```

File: src/RulesMD/RulesMD.csproj [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+<Target Name="GenerateMarkdown" AfterTargets="Build" Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU'">
```
Comment (DaveTryon) [inline src/RulesMD/RulesMD.csproj]: Indent is off
```suggestion
  <Target Name="GenerateMarkdown" AfterTargets="Build" Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU'">
```

File: src/RulesMD/RulesMD.csproj [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  <Exec Command="echo Generate markdown of rules info" />
```
Comment (DaveTryon) [inline src/RulesMD/RulesMD.csproj]: Indent is off
```suggestion
    <Exec Command="echo Generate markdown of rules info" />
```

File: src/RulesMD/RulesMD.csproj (hunk 6 above)
Comment (DaveTryon) [inline src/RulesMD/RulesMD.csproj]: Indent is off
```suggestion
    <Exec Command="$(TargetPath) --output $(TargetDir)\axe-windows-rules.md"  />
```

File: src/RulesMD/Helpers.cs (hunk 4 above)
Comment (RobGallo) [reply src/RulesMD/Helpers.cs]: The hard-coded strings were a hold-over from when this project wasn't included in the axe-windows solution. But now the change has been made to use the RuleSelection resources.
